File: Algorithms/Recursive Algorithms/C/GCD.c
#include <stdio.h>
int GCD(int, int);
int main(){
	int num1, num2;
	printf("Enter a number to find GCD: ");
	scanf("%d", &num1);
	printf("Enter another number to find GCD: ");
	scanf("%d", &num2);
	printf("GCD of %d and %d : %d", num1, num2, GCD(num1, num2));
}
int GCD(int a, int b){
	if(a == 0 || b == 0){
		return 0;
	}else if(a == b){
		return a;
	}else if(a > b){
		return GCD(a-b,b);
	}else if(b > a){
		return GCD(a,b-a);
	}
}

File: Algorithms/Recursive Algorithms/Go/GCD.go
package main

import "fmt"

// simple solution -- substracting smaller number from larger
func GCD1(a, b int) int {
	if a == 0 {
		return b
	}
	if b == 0 {
		return a
	}
	if a == b {
		return a
	}

	if a > b {
		return GCD1(a-b, b)
	} else {
		return GCD1(a, b-a)
	}
}

// Euclidean algorithm
func GCD2(a, b int) int {
	if b == 0 {
		return a
	}
	return GCD2(b, a%b)
}

func main() {

	var first, second int

	fmt.Print("Enter first  number : ")
	fmt.Scanln(&first)
	fmt.Print("Enter second number : ")
	fmt.Scanln(&second)

	fmt.Print("GCD using simple solution : ")
	fmt.Println(GCD1(first, second))
	fmt.Print("GCD using Euclidean Algorithm : ")
	fmt.Println(GCD2(first, second))

}
